src/environment.go

```diff
@@ -56,6 +56,21 @@ type cache interface {
 	set(key, value string, ttl int)
 }
 
+type windowsRegistryValueType int
+
+const (
```

Comment:
should we move these to `environment_windows` so they don't get added for other platforms?

Reply:
I was wondering about that too - if possible to have the same interface, but on windows the windowsRegistryValueType and interface etc is full featured where on other platforms it's an interface with no info... I'll have a try.

Reply:
ah - problem with that, segment_winreg.go has a dependency on that interface and enum now, so to make it windows-only at an environment level means the segment would then have to have different platform impls to avoid using the interface/enum.

Reply:
Gotcha!